ot/lp/__init__.py

```diff
@@ -156,17 +177,30 @@ def emd2(a, b, M, processes=multiprocessing.cpu_count(), numItermax=100000):
 
     # if empty array given then use unifor distributions
     if len(a) == 0:
-        a = np.ones((M.shape[0], ), dtype=np.float64)/M.shape[0]
+        a = np.ones((M.shape[0],), dtype=np.float64) / M.shape[0]
     if len(b) == 0:
-        b = np.ones((M.shape[1], ), dtype=np.float64)/M.shape[1]
+        b = np.ones((M.shape[1],), dtype=np.float64) / M.shape[1]
 
-    if len(b.shape) == 1:
-        return emd2_c(a, b, M, numItermax)
+    if log:
+        def f(b):
+            G, cost, u, v, resultCode = emd_c(a, b, M, num_iter_max)
+            result_code_string = check_result(resultCode)
+            log = {}
+            log['G'] = G
```

Comment:
i would remove G here because it does not scale well in memory when computing the distance between a lot of samples .

Maybe add a parameter return_matrix to the function?